chemprop/cli/hyperopt.py

```diff
@@ -1,16 +1,348 @@
+                                build_splits, normalize_inputs,
+                                process_train_args, validate_train_args)
 from chemprop.cli.utils.command import Subcommand
+from chemprop.data import MolGraphDataLoader
+from chemprop.nn import AggregationRegistry
+from chemprop.nn.utils import Activation
+
+
+logger = logging.getLogger(__name__)
+
+AVAILABLE_SPACES = {
+    "activation": tune.choice(list(Activation.keys())),
+    "aggregation": tune.choice(list(AggregationRegistry.keys())),
+    "aggregation_norm": tune.quniform(lower=1, upper=200, q=1),
+    "batch_size": tune.quniform(lower=5, upper=200, q=5),
+    "depth": tune.randint(lower=2, upper=6),
+    "dropout": tune.choice(
+        [
+            tune.choice([0.]),
+            tune.quniform(lower=0.05, upper=0.4, q=0.05),
+        ],
+    ),
+    "ffn_hidden_size": tune.quniform(lower=300, upper=2400, q=100),
+    "ffn_num_layers": tune.randint(lower=2, upper=6),
+    "final_lr_ratio": tune.loguniform(lower=1e-4, upper=1),
+    "hidden_size": tune.quniform(lower=300, upper=2400, q=100),
+    "init_lr_ratio": tune.loguniform(lower=1e-4, upper=1),
+    "linked_hidden_size": tune.quniform(lower=300, upper=2400, q=100),
+    "max_lr": tune.loguniform(lower=1e-6, upper=1e-2),
+    "warmup_epochs": None
+}
+
+SEARCH_PARAM_KEYWORDS_MAP = {
+    "basic": ["depth", "ffn_num_layers", "dropout", "linked_hidden_size"],
+    "learning_rate": ["max_lr", "init_lr", "final_lr", "warmup_epochs"],
+    "all": list(AVAILABLE_SPACES.keys()),
+}
 
+def get_available_spaces(train_epochs: int) -> dict:
+
+    AVAILABLE_SPACES["warmup_epochs"] = tune.quniform(lower=1, upper=train_epochs // 2, q=1)
+
+    return AVAILABLE_SPACES
 
 class HyperoptSubcommand(Subcommand):
     COMMAND = "hyperopt"
     HELP = "perform hyperparameter optimization on the given task"
 
     @classmethod
     def add_args(cls, parser: ArgumentParser) -> ArgumentParser:
-        pass
+        parser = add_common_args(parser)
+        parser = add_train_args(parser)
+        return add_hyperopt_args(parser)
 
     @classmethod
     def func(cls, args: Namespace):
-        pass
+        args = process_common_args(args)
+        args = process_train_args(args)
+        args = process_hyperopt_args(args)
+        validate_common_args(args)
+        validate_train_args(args)
+        main(args)
+
+def add_hyperopt_args(parser: ArgumentParser) -> ArgumentParser:
+
+    hyperopt_args = parser.add_argument_group("Hyperparameter optimization arguments")
+
+    hyperopt_args.add_argument(
+        "--num-samples",
+        type=int,
+        default=10,
+        help="Number of hyperparameter optimization samples to run",
+    )
+
+    hyperopt_args.add_argument(
+        "--hyperopt-seed",
+        type=int,
+        default=0,
+        help="The initial seed used for choosing parameters in hyperparameter optimization trials. In each trial, the seed will be increased by one, skipping seeds previously used.",
```

Comment:
It is not used.

Reply:
I renamed this to `--hyperopt-random-state-seed` and passed it into `HyperOptSearch`